### PR TITLE
Add generate randomness

### PR DESCRIPTION
**Pull request by kim201212**

### CODE REVIEW

**Review by rootwarp — COMMENTED** [review]
Did you test this?

**Comment by kim201212** [comment]
close #15

File: contribution/contribution.js [hunk 1]
```diff
@@ -10,20 +28,18 @@ function contribute(contributions, rand) {
...
-    const Fr = bls.bls12_381.CURVE.Fr;
 
     const util = bls.bls12_381.utils;
 
     for(var i = 0; i < contributions.length; i++) {
         const g1Powers = contributions[i].powersOfTau.G1Powers;
         const g2Powers = contributions[i].powersOfTau.G2Powers;
-
-        var xi = 1n;
+        var xi = BigInt(1);
```
Comment (rootwarp) [inline contribution/contribution.js:37]: afaik, this is same to previous one. why we should change to this?

File: index.js [hunk 2]
```diff
@@ -70,20 +71,23 @@ program
...
             break;
         }
 
         console.log('Decoding...');
         contributions = conversion.decode(resp.contributions);
+        //contributions = conversion.decode('{"contributions": [ '+resp.contributions+'}');
 
-        const randValue = Math.floor(Math.random() * 100000); // TODO:
+        rand = contribute.generateRandom();
+        rand = Fr.create(rand);
```
Comment (rootwarp) [inline index.js:82]: What this means?

**Review by rootwarp — COMMENTED** [review]
May `./test/dump.json` should be removed.

File: contribution/contribution.js (hunk 1 above)
Comment (kim201212) [reply contribution/contribution.js:37]: It will help to distinguish it clearly.

**Comment by rootwarp** [comment]
In my env. error occurs.

```
rootwarp@LT-026 ~/g/d/dsrv-kzg-ceremony random > ./index.js ceremony a002479f-782b-48a5-b3b9-fa18c87446f1
Starting ceremony...
Decoding...
Update Power of Tau...
updatePowerOfTau 7177057640311544215210810364117382200091468669765550551577723128495801892862n
/Users/rootwarp/git/dsrv/dsrv-kzg-ceremony/node_modules/@noble/curves/lib/abstract/weierstrass.js:113
            throw new Error('Expected valid bigint: 0 < bigint < curve.n');
                  ^

Error: Expected valid bigint: 0 < bigint < curve.n
    at assertGE (/Users/rootwarp/git/dsrv/dsrv-kzg-ceremony/node_modules/@noble/curves/lib/abstract/weierstrass.js:113:19)
    at Point.multiply (/Users/rootwarp/git/dsrv/dsrv-kzg-ceremony/node_modules/@noble/curves/lib/abstract/weierstrass.js:404:13)
    at Object.contribute (/Users/rootwarp/git/dsrv/dsrv-kzg-ceremony/contribution/contribution.js:42:46)
    at Command.<anonymous> (/Users/rootwarp/git/dsrv/dsrv-kzg-ceremony/index.js:85:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v19.6.0
```


As I debug the code, 

`contribute.contribute(contributions, rand)` function takes just single value of `rand` argument, not array type. did you check this?

659f691

File: contribution/contribution.js (hunk 1 above)
Comment (rootwarp) [reply contribution/contribution.js:37]: 👌

File: index.js (hunk 2 above)
Comment (kim201212) [reply index.js:82]: We can get a valid number if we put a number that satisfies fr.

Comment (rootwarp) [reply index.js:82]: 👌

**Comment by kim201212** [comment]
@rootwarp 

https://github.com/dsrvlabs/dsrv-kzg-ceremony/pull/9/commits/494b8d6110769e4d475eae627880dfed1c0ea5b9

I fixed it at the above commit and tested it. 🙏

**Review by rootwarp — APPROVED** [review]
LGTM

**Comment by rootwarp** [comment]
> @rootwarp
> 
> [494b8d6](https://github.com/dsrvlabs/dsrv-kzg-ceremony/commit/494b8d6110769e4d475eae627880dfed1c0ea5b9)
> 
> I fixed it at the above commit and tested it. 🙏

@kim201212 Branch seems conflicted. can you check this?

**Comment by rootwarp** [comment]
@kim201212 As I remember this code already merged.
This should be close.

**Comment by kim201212** [comment]
It already merged, Close

**Pull request closed without merging**